### PR TITLE
Add ssamadgh/ModelAssistant to suite

### PR DESCRIPTION
**Pull request by ssamadgh**
### Pull Request Description

Added ModelAssistant Repository to **Projects.json**

### Acceptance Criteria

To be accepted into the Swift source compatibility test suite, a project must:

- [x] be an *Xcode* or *swift package manager* project
- [x] support building on either Linux or macOS
- [x] target Linux, macOS, or iOS/tvOS/watchOS device
- [x] be contained in a publicly accessible git repository
- [x] maintain a project branch that builds against Swift 4.0 and passes any unit tests
- [x] have maintainers who will commit to resolve issues in a timely manner
- [x] be compatible with the latest GM/Beta versions of *Xcode* and *swiftpm*
- [x] add value not already included in the suite
- [x] be licensed with one of the following permissive licenses:
	* MIT
- [x] pass `./project_precommit_check` script run

Ensure project meets all listed requirements before submitting a pull request.

### CODE REVIEW

**Comment by clackary** [comment]
@ssamadgh Thanks for opening this! Allow me some time to review, and I'll get testing started when review is complete.

File: projects.json [hunk 1]
```diff
@@ -1223,6 +1223,31 @@
...
+        "version": "4.2",
+        "commit": "1f1d37da8e3161a374b25f0e8ca9c3f22b352b7f"
+      }
+    ],
+    "platforms": [
+      "Darwin"
+    ],
+    "actions": [
+      {
+        "action": "BuildXcodeWorkspaceScheme",
```
Comment (clackary) [inline projects.json]:
```suggestion
        "action": "BuildXcodeProjectScheme",
```

File: projects.json [hunk 2]
```diff
@@ -1223,6 +1223,31 @@
...
+        "workspace": "ModelAssistant.xcodeproj",
```
Comment (clackary) [inline projects.json]:
```suggestion
        "project": "ModelAssistant.xcodeproj",
```

**Comment by clackary** [comment]
@swift-ci test

**Comment by clackary** [comment]
@swift-ci test

**Comment by clackary** [comment]
@swift-ci test

**Comment by ssamadgh** [comment]
@clackary I have no idea how to fix Swift-ci test errors. Is there any guidelines for this?

**Comment by ssamadgh** [comment]
@clackary @swift-ci Please test source compatibility.

**Comment by clackary** [comment]
@swift-ci test

**Comment by clackary** [comment]
@swift-ci test

**Comment by clackary** [comment]
@ssamadgh Yes, I'd be happy to test again. The only way to start testing for this is for someone in the Swift Committers group to use `@swift-ci test`. Don't worry about updating the title.

Looking at the most recent failure, this may be a compiler crash. Let's see if it still happens after your changes to the library.

**Comment by clackary** [comment]
Looks like this is crashing in the same way as before. I've filed [SR-9102](https://bugs.swift.org/browse/SR-9102) to investigate. For the meantime, we can xfail this to be able to get it merged.

File: projects.json [hunk 3]
```diff
@@ -1223,6 +1223,31 @@
...
+    "platforms": [
+      "Darwin"
+    ],
+    "actions": [
+      {
+        "action": "BuildXcodeProjectScheme",
+        "project": "ModelAssistant.xcodeproj",
+        "scheme": "ModelAssistant iOS",
+        "destination": "generic/platform=iOS",
+        "configuration": "Release"
```
Comment (clackary) [inline projects.json]:
```suggestion
        "configuration": "Release",
        "xfail": {
          "compatibility": {
            "4.2": {
              "branch": {
                "master": "https://bugs.swift.org/browse/SR-9102"
              }
            }
          }
        }
```

**Comment by ssamadgh** [comment]
@clackary I committed the changes. @swift-ci test.

**Comment by clackary** [comment]
@swift-ci test

**Comment by clackary** [comment]
@swift-ci test

**Comment by clackary** [comment]
@ssamadgh Thanks for setting this up!

**Comment by ssamadgh** [comment]
@clackary Thanks for merging. What should I do if I updated my repository?
Is it enough to update the commit id?

**Pull request merged**